sky/optimizer.py

```diff
@@ -846,12 +885,13 @@ def _fill_in_launchable_resources(
                     f'sky check {colorama.Style.RESET_ALL}, or change the '
                     'cloud requirement')
         elif resources.is_launchable():
+            launchable[resources] = _generate_launchables_with_region_zones(
+                resources)
             if isinstance(resources.cloud, clouds.GCP):
                 # Check if the host VM satisfies the max vCPU and memory limits.
-                clouds.GCP.check_accelerator_attachable_to_host(
-                    resources.instance_type, resources.accelerators,
-                    resources.zone)
-            launchable[resources] = [resources]
+                for r in launchable[resources]:
+                    clouds.GCP.check_accelerator_attachable_to_host(
+                        r.instance_type, r.accelerators, r.zone)
```

Comment:
Good point. I changed the order so that we check the host-accelerator compatibility before making region/zone-specified launchables. If some zones do not pass the check while others do,
1) If the user has specified the zone and the zone does not support the host-accelerator combination, then it raises error,
2) If the user has not specified the zone, launchables will be made for all zones even if there are some zones that do not support the host-accelerator combination.

Reply:
Case (2) seems to refer to https://github.com/skypilot-org/skypilot/blob/844a5c936af1e58a630053f0154ab0e8b4599d9b/sky/optimizer.py#L911-L912 this list including some invalid zones. Are we saying that we permit them to be tried by the provisioner, which should correctly return an error and failover will kick in? Worth a comment.

Reply:
Yes. I think it's fine because that case happens very rarely; only when the user requested GCP 4:P100 or 8:K80 with a very large host VM.

The relevant piece of code in `gcp_catalog.py#check_accelerator_attachable_to_host()`:
```python
    if acc_name == 'K80' and acc_count == 8:
        if zone in ['asia-east1-a', 'us-east1-d']:
            max_memory = 416
    elif acc_name == 'P100' and acc_count == 4:
        if zone in ['us-east1-c', 'europe-west1-d', 'europe-west1-b']:
            max_cpus = 64
            max_memory = 208
```

Added a comment on this. PTAL.